### PR TITLE
[#157340066] Load all asset make dropdown asset types

### PR DESCRIPTION
**Pull request by Ewanjiru**
## What does this PR do?
Temporarily fixes the dropdown pagination issue for asset types on asset make modal component.

## Description of Task to be completed?
Enable retrieving of all asset types to be loaded on the dropdown by bypassing the pagination.

## How should this be manually tested?
On the assets page, click on the + icon for asset makes and on the displayed modal click on the asset types drop down. This should show all existing asset types.

## What are the relevant pivotal tracker stories?
157340066

## Any background context you want to add?
N/A
## Important notes
N/A
## Packages installed
N/A
## Deployment note
N/A
## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation
## Screenshots
screen shot 2018-07-10 at 15 39 19

### CODE REVIEW

File: src/components/common/ModalComponent.jsx [hunk 1]
```diff
@@ -20,14 +20,14 @@ export default class ArtModal extends Component {
       <div className={this.props.className}>
         <Modal
           trigger={
-            (
-              <span tabIndex="-1" role="button" onClick={this.toggleModal} onKeyUp={(() => {})}>
-                { this.props.trigger }
+            this.props.trigger ? (
+              <span tabIndex="-1" role="button" onClick={this.toggleModal} onKeyUp={(() => { })}>
```
Comment (sachungo) [inline src/components/common/ModalComponent.jsx]: OnKeyUp value should be `onKeyUp={() => {}}`. Note that there is no empty space between the return object of `() => {}`

File: src/components/common/ModalComponent.jsx [hunk 2]
```diff
@@ -20,14 +20,14 @@ export default class ArtModal extends Component {
...
+                {this.props.trigger}
               </span>
-            ) || (<i
+            ) : (<i
               className="plus link icon"
               onClick={this.toggleModal}
-              onKeyUp={() => {}}
+              onKeyUp={() => { }}
```
Comment (sachungo) [inline src/components/common/ModalComponent.jsx]: Same comment above applies here.

Please restructure the trigger block to:
```
this.props.trigger ? (
  <span
    tabIndex="-1"
    role="button"
    onClick={this.toggleModal}
    onKeyUp={() => {}}
  >
    {this.props.trigger}
  </span>
) : (
  <i
     className="plus link icon"
     onClick={this.toggleModal}
     onKeyUp={() => {}}
     role="button"
     tabIndex="-1"
  />
)}

File: src/_actions/assetTypes.actions.js [hunk 3]
```diff
@@ -40,3 +42,45 @@ export const createAssetTypeSuccess = assetType => (
 export const createAssetTypeFailure = error => (
   { type: CREATE_ASSET_TYPE_FAILURE, payload: error }
 );
+
+export const loadDropdownAssetTypes = () => (dispatch) => {
+  const pageNumber = 1;
+  axios
+    .get(`asset-types?page=${pageNumber}`)
+    .then((response) => {
+      const pages = Array.from(
```
Comment (sachungo) [inline src/_actions/assetTypes.actions.js]: Once you find time, please look for me we look at this

File: src/components/common/ModalComponent.jsx (hunk 2 above)
Comment (Ewanjiru) [reply src/components/common/ModalComponent.jsx]: Noted

File: src/_actions/assetTypes.actions.js (hunk 3 above)
Comment (Ewanjiru) [reply src/_actions/assetTypes.actions.js]: Noted. I will.

Comment (Ewanjiru) [reply src/_actions/assetTypes.actions.js]: @sachungo  As discussed, we will keep this implementation as we wait for the pagination correction in the backend.

File: src/components/common/ModalComponent.jsx (hunk 1 above)
Comment (Ewanjiru) [reply src/components/common/ModalComponent.jsx]: Done

File: src/_actions/assetTypes.actions.js [hunk 4]
```diff
@@ -40,3 +42,44 @@ export const createAssetTypeSuccess = assetType => (
...
+          });
+      });
+    })
+    .catch((error) => {
+      dispatch(dropdownAssetTypeFailure(error));
+      dispatch(updateToastMessageContent(`Could not load the asset types. ${error}`, 'error'));
+    });
+};
+
+export const dropdownAssetTypeSuccess = allAssetTypes => (
```
Comment (sachungo) [inline src/_actions/assetTypes.actions.js]: Nitpick: reorder this according to the snippet
```
export const dropdownAssetTypeSuccess = allAssetTypes => ({
  type: LOAD_DROPDOWN_ASSET_TYPES_SUCCESS,
  payload: allAssetTypes
});
```

File: src/_actions/assetTypes.actions.js [hunk 5]
```diff
@@ -40,3 +42,44 @@ export const createAssetTypeSuccess = assetType => (
...
+  { type: LOAD_DROPDOWN_ASSET_TYPES_SUCCESS, payload: allAssetTypes }
+);
+
+export const dropdownAssetTypeFailure = error => (
```
Comment (sachungo) [inline src/_actions/assetTypes.actions.js]: Also apply the above nitpick here

Comment (Ewanjiru) [reply src/_actions/assetTypes.actions.js]: Well noted. Thanks

**Pull request merged**